website/server/libs/cron.js

```diff
@@ -308,6 +308,12 @@ export function cron (options = {}) {
         atLeastOneDailyDue = shouldDo(thatDay.toDate(), task, user.preferences);
       }
     } else {
+      // add history entry when task was not completed
+      task.history.push({
+        date: Number(new Date()),
+        value: task.value,
+      });
+
```

Comment:
This still needs to be done after all the code below because some of that code will affect the value. For example, if Stealth doesn't protect the Daily, then scoreTask around line 340 will make the Daily redder, and it's that redder value that we'd want to record. Probably putting this code at the bottom of the `else` block is what we want.

Reply:
Oh okok got it. 
Sorry for all the changes, I hope to catch up quicker next time.

Reply:
No worries! :) The cron code is large and complex.